model/src/main/java/userinfo/UserInfo.java

```diff
@@ -0,0 +1,42 @@
+package userinfo;
+
+/**
+ * Stores the information of a user.
+ */
+public abstract class UserInfo {
+    private String address;
+    private String phoneNumber;
+
+    /**
+     * Constructs a UserInfo object.
+     *
+     * @param address: the address of the user
+     * @param phoneNumber: the phone number of the user
+     */
+    public UserInfo(String address, String phoneNumber){
+        this.address = address;
+        this.phoneNumber = phoneNumber;
+    }
+
+    public String getAddress() {
+        return address;
+    }
+
+    public void setAddress(String address) {
+        this.address = address;
+    }
+
+    public String getPhoneNumber() {
+        return phoneNumber;
+    }
+
+    public void setPhoneNumber(String phoneNumber) {
+        this.phoneNumber = phoneNumber;
+    }
+
+    /**
+     * Gets the type of the user.
+     * @return "CLIENT" or "PROVIDER" depending on the type of this user
+     */
+    public abstract String getType();
```

Comment:
Instead of returning a `String` here, it might make more sense to return an enum of some kind as laid out in the CRC model. A `UserType` enum with `CLIENT` and `PROVIDER` fields could be created.